.github/workflows/test-stellar.yaml

```
@@ -0,0 +1,79 @@
+name: Test Stellar
+
+on: pull_request
+
+jobs:
+  check-relevant-changes:
+    name: Check for Relevant Changes
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    outputs:
+      run_tests: ${{ steps.filter.outputs.stellar == 'true' || steps.filter.outputs.common == 'true' || steps.filter.outputs.github == 'true' }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            stellar:
+              - 'stellar/**'
+            common:
+              - 'common/**'
+            github:
+              - '.github/workflows/test-stellar.yaml'
+      - name: Summarize Changes
+        run: |
+          echo "Changes in stellar: ${{ steps.filter.outputs.stellar }}"
+          echo "Changes in common: ${{ steps.filter.outputs.common }}"
+          echo "Changes in github: ${{ steps.filter.outputs.github }}"
+
+  test-stellar:
+    name: Test Stellar
+    needs: check-relevant-changes
+    if: ${{ needs.check-relevant-changes.outputs.run_tests == 'true' }}
+    runs-on: blacksmith-8vcpu-ubuntu-2204
+    steps:
+      # this sets up nested repos: https://github.com/actions/checkout?tab=readme-ov-file#checkout-multiple-repos-nested
+      # axelar-contract-deployments
+      #   - axelar-amplifier-stellar
+      #
+      # future steps depend on this directory structure
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Checkout axelar-amplifier-stellar repo
+        uses: actions/checkout@v4
+        with:
+          repository: axelarnetwork/axelar-amplifier-stellar
+          path: axelar-amplifier-stellar
+
+      - name: Get Latest Short Commit Hash from axelar-amplifier-stellar
+        id: get_short_commit_hash
+        run: |
+          cd axelar-amplifier-stellar
+          git fetch --all
```

```suggestion
      - name: Checkout code
        uses: actions/checkout@v4

      - name: Checkout axelar-amplifier-stellar repo
        uses: actions/checkout@v4
        with:
          repository: axelarnetwork/axelar-amplifier-stellar
          path: axelar-amplifier-stellar

      - name: Get latest commit for axelar-amplifier-stellar
        id: commit_hash
        run: |
          cd axelar-amplifier-stellar
          git fetch --all
```
ideally steps are readable enough that comments aren't needed

---

make sense, updated.